Fix/#3110 fixed vmstat data points import

Patrykb0802 opened the pull request.

Patrykb0802 committed:
#3110 Fixed VMstat data points import

Patrykb0802 committed:
Revert "#3110 Fixed VMstat data points import"

This reverts commit 630dedee09dd9e5912efc32f64355dffba1037fd.

Patrykb0802 committed:
#3110 Fixed VMstat data points import

Limraj reviewed (commented):
Improved importing of Attribute parameter:
![image](https://github.com/user-attachments/assets/3d9b0cd7-3bc4-4048-aaf6-6c50c58743e5)

But not import data point properties:
![image](https://github.com/user-attachments/assets/22ac0239-8234-4c4d-bb08-d6e8017e5ec7)

I recommend exporting the VMStat Data Source, then importing it, copying the json from the json_project.txt file and importing it using the function:
![image](https://github.com/user-attachments/assets/ce4e1d7d-ebd0-4b9e-ac2f-2d866ce1a13f)

and look at the errors in the import:
`class com.serotonin.json.JsonNumber cannot be cast to class com.serotonin.json.JsonString (com.serotonin.json.JsonNumber and com.serotonin.json.JsonString are in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @7cd3860)`
![image](https://github.com/user-attachments/assets/7ed6e22d-2415-464c-bd0e-179207d26da0)

Patrykb0802 committed:
#3110 fixed vmstat data points import

Patrykb0802 committed:
#3110 fixed vmstat data points import

Limraj commented on src/com/serotonin/mango/vo/dataSource/vmstat/VMStatPointLocatorVO.java:
```diff
@@ -155,10 +155,10 @@ private void readObject(ObjectInputStream in) throws IOException {
     @Override
     public void jsonDeserialize(JsonReader reader, JsonObject json) throws JsonException {
-        String text = json.getString("attributeId");
+        String text = json.getInt("attributeId").toString();
```
You probably don't need to add special handling for this field since there is a annotation next to it. (@JsonRemoteProperty)

Patrykb0802 committed:
#3110 fixed vmstat data points import

Limraj committed:
Merge remote-tracking branch 'origin/release/2.8.0' into fix/#3110_Fixed_VMstat_data_points_import

Limraj committed:
#3110 Fixed vmstat data points import:
- Added ExportImportJsonUtils.getInt;
- Removed annotation @JsonRemoteProperty from attributeId;
- Added implementation for method VMStatPointLocatorVO.jsonDeserialize with use ExportImportJsonUtils.getInt;
